.changeset/bitter-cougars-deny.md

```diff
@@ -0,0 +1,8 @@
+---
+'@primer/react': patch
+---
+
+AvatarStack:
+
+* Border was incorrectly applying to elements that were not `Avatar` children.
+* The square prop was not applying to individual `Avatar` components.
```

Comment:
Just a thought, the way this shows up in our changelog it could be helpful to split this up into two changesets